src/GridFS/StreamWrapper.php

```diff
@@ -324,6 +295,14 @@ public function stream_write(string $data): int
         return $this->stream->writeBytes($data);
     }
 
+    public function unlink(string $path): bool
+    {
+        $context = $this->getContext($path, 'r');
```

Comment:
`Bucket::resolveStreamContext()` executes `findFileByFilenameAndRevision()` for read modes in order to fetch the file document. Would it make sense to use `'w'` here and avoid the unnecessary query?

I understand `rename()` needs to use `'r'` since it does read the filename from the file document; however, there may be a refactoring opportunity there if you'd just like to trust the `$fromPath` string. I suppose it's a matter of intentionally NOP-ing on a FileNotFoundException or executing an update query that may not match anything -- so I'd be find leaving `rename()` as-is.

Reply:
The idea of using `r` mode was to check that the file exists before doing operations, to throw a `FileNotFoundException`. This could be optimized by checking the number of updated documents after the update, but that's not feasible if `Write Concern = 0`.

Reply:
I reverted to throw a `FileNotFoundException` in `rename` and `unlink`, instead of silently returning `false`. This behavior is consistent with S3 stream wrapper.